com.github.maoschanz.DynamicWallpaperEditor.json

```diff
@@ -0,0 +1,48 @@
+{
+    "app-id" : "com.github.maoschanz.DynamicWallpaperEditor",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.28",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "dynamic-wallpaper-editor",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=home:rw"
```

Comment:
I think we should try to tighten this down to `xdg-pictures` only.

Reply:
it's for opening/saving XML files, and i don't know where people put them

Reply:
Your app should use GtkFileChooserNative for open and save dialogs. Then you don't need direct read/write access.

If there is a single path you need to auto export a file for like GNOME-Shell or such then you should only add enough for that.

Reply:
Gio.File.new_for_path(self.xml_file_name)
            f = open(self.xml_file_name, 'w')
            f.write(self.generate_text())
            f.close()

That sounds like direct access: whatever is the dialog i use for getting the file name, the app is reading, parsing, generating, writing text from XML files

Reply:
You show a popup dialog to ask the user where to save it right? Well use `Gtk.FileChooserNative.new()` to make that dialog. Then get your `Gio.File` from that.

https://lazka.github.io/pgi-docs/#Gtk-3.0/classes/FileChooserNative.html#Gtk.FileChooserNative.new

Reply:
Amazing, thank you, with all these classes and interfaces about chosing files i missed that i could get directly a gio file. Does it mean i can provide no access to any part of the filesystem and it will work like currently ?

Reply:
I've not used the app to see how it actually works, it is possible if all file interactions require the user select a file first.

Reply:
Sadly it doesn't work very well, i can open XML files, but then it's hard to load their content (which is a list of paths to pictures)

Reply:
I think i will use `--filesystem=home:ro`, is it fine ?